services/sonar/sonar-generic.tester.js

```diff
@@ -0,0 +1,11 @@
+'use strict'
+
+const { isMetric } = require('../test-validators')
+const t = (module.exports = require('../tester').createServiceTester())
+
+t.create('Security Rating')
+  .get('/https/sonarcloud.io/com.luckybox:luckybox/security_rating.json')
+  .expectBadge({
+    label: 'security rating',
+    message: isMetric,
+  })
```

Comment:
Debated whether I should add generic tests like this one for the other 60+ metrics. I decided against it adding them here as part of this PR since it was really just supposed to be a refactor. 

Sonar code coverage was < 50% beforehand, but is now 100% with tests for all of the metrics we provide first class badges for (the ones in the other service classes) so I think we're going to be in a much better place Sonar validation wise. 

I'd rather add tests for the other metrics when they are enhanced as part of #3236